MapboxDirections/MBDirections.swift

```diff
@@ -272,10 +287,17 @@ open class Directions: NSObject {
      After requesting the URL returned by this method, you can parse the JSON data in the response and pass it into the `Route.init(json:waypoints:profileIdentifier:)` initializer.
      */
     @objc(URLForCalculatingDirectionsWithOptions:)
-    open func url(forCalculating options: DirectionOptions) -> URL {
-        let params = options.params + [
-            URLQueryItem(name: "access_token", value: accessToken),
-        ]
+    open func url(forCalculating options: DirectionsOptions) -> URL {
+        let params: [URLQueryItem]
+        if options is MatchingOptions {
```

Comment:
Directions shouldn’t have any code that switches on the type of `options` like this. This conditional seems to exist for the purpose of ignoring the `options.params` property. That’s surprising. It suggests that we need to push more functionality into RouteOptions and MatchingOptions rather than working around the functionality here in Directions.